### PR TITLE
Visual fixes for new merge datasets

### LINKED ISSUE #4277: Merge tables fixes
- [x] **column join** and **spatial join** `#666` header icons

- [x] Delete footer line:
artboard

- [x] Remove the scroll-to-top on selecting a column
- [x] Do not re-sort collection on selection
- [x] Column merge: Only show the connector on the 2nd step
- [x] Spatial join: remove the "merge column" connector line (the lower one)

- [x] Change `Select the columns you want to add as well.` copy to `Define the rest columns you want in your dataset`

- [x] Change `First, select the column you want to merge in both datasets. You can only merge datasets by joining by columns of the same type (e.g. number to a number).`
copy to
`Select the dataset on the right that you want to merge the left with. You can only merge datasets by joining by columns of the same type (e.g. number to a number).`

- [x] Change header copy to `Dataset merging is useful if you want to combine data from two datasets into a single new dataset.`
screen shot 2015-07-10 at 7 17 42 pm

- [x] Change all column list spacings to `12px` **column join** and **space join**:
screen shot 2015-07-13 at 12 20 05 pm

- [x] Delete the line on `Add remaining column` step boxes and resize it to `115px` **column join** and **space join**:
screen shot 2015-07-10 at 12 35 57 pm

- [x] Align radio buttons to middle label line **column join** and **space join**:
screen shot 2015-07-13 at 12 01 43 pm

- [x] **column join** and **spatial join** fetching data state
screen shot 2015-07-02 at 4 00 04 pm

- [x] **column join** and **spatial join** not-compatible column (show tooltip for why can't be selectable):
screen shot 2015-07-02 at 4 00 14 pm

---

- [x] **column join** sticked header:
sticked header copy 2

### PR DESCRIPTION
**Pull request by viddo**
Fixes #4277, see the issue for scope
screen shot 2015-07-15 at 13 23 37
screen shot 2015-07-15 at 13 23 41
screen shot 2015-07-15 at 13 23 53
screen shot 2015-07-15 at 13 23 58
screen shot 2015-07-15 at 13 27 33

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4986/))

**Comment by viddo** [comment]
@javierarce can you review this one please? It's probably easiest if you checkout the branch and test locally (since all stagings are occupied anyway
cc @josecruz  for visuals

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4987/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5008/))

**Comment by viddo** [comment]
@javierarce bump for review! ;)
deployed on ded03 for testing

**Comment by josecruz** [comment]
Yay! Good work @viddo. Just found this bugs:

- [ ] This spacings from dropdowns/boxes to list should be `12px` (this happens in column merge too):
artboard1

- [ ] This should be aligned to middle-line:
artboard

- [ ] Margings here should be `28px`: 
artboard2

- [ ] Header icons should be `#666`

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5014/))

File: lib/assets/javascripts/cartodb/common/dialogs/merge_datasets/column_merge/select_columns_model.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ var MergeModel = require('../merge_step_model');
  */
 module.exports = cdb.core.Model.extend({
 
-  instructions: 'Select the columns you want to add as well.',
+  instructions: 'Define the rest columns you want in your dataset.',
```
Comment (javierarce) [inline lib/assets/javascripts/cartodb/common/dialogs/merge_datasets/column_merge/select_columns_model.js]: I'd say: "Choose other columns you want in your dataset." or "Pick other columns you want in your dataset"

File: lib/assets/javascripts/cartodb/common/dialogs/merge_datasets/merge_datasets_content.jst.ejs [hunk 2]
```diff
@@ -26,18 +26,18 @@
...
         <% } else { %>
           <li class="CreateDialog-headerStep CreateDialog-headerStep--single is-selected">
-            <div class="Dialog-headerIcon">
+            <div class="Dialog-headerIcon Dialog-headerIcon--neutral">
               <i class="iconFont iconFont-MergeArrow"></i>
             </div>
             <p class="Dialog-headerTitle">Merge with another table</p>
-            <p class="Dialog-headerText">Table merging is useful if you want to combine data from two tables into a single new table.</p>
-            <p class="Dialog-headerText">You can merge tables by a column attribute or as a spatial intersection.</p>
+            <p class="Dialog-headerText">Dataset merging is useful if you want to combine data from two datasets into a single new dataset</p>
```
Comment (javierarce) [inline lib/assets/javascripts/cartodb/common/dialogs/merge_datasets/merge_datasets_content.jst.ejs]: We repeat data four times. I'd say: "Merging is useful if you want to combine data from two datasets into a single new one".

**Comment by javierarce** [comment]
:+1: code looks fine. I just added a couple of comments regarding the texts.

**Comment by viddo** [comment]
Thx! Fixed all commented issues, so once green I'll go ahead and merge

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5031/))

**Comment by javierarce** [comment]
:dancer:

**Comment by viddo** [comment]
FYI, reverted one code-review fix for the margins, so the selected items have the same size as the rest and the extra 12px spacing between select and the list items is back, to avoid the weird look and jumpy behavior. See:
margins4

(Btw, the "selected-stuff" boxes are actually white, they appear wrong due to the gif color palette…)

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5037/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5038/))

**Comment by josecruz** [comment]
Found this:

When you go back, all columns appears selected:
untitled

Align `Back` and `Close` buttons to the steps icons:
screen shot 2015-07-16 at 12 36 05 pm

Count icon:
screen shot 2015-07-16 at 12 44 33 pm

**Comment by viddo** [comment]
OK, so that should be all, I hope. re-deployed on ded03, but should be good to go now

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5042/))

**Comment by viddo** [comment]
The items in the original issue + the coder review items are now resolved, so if there are more bugs or feature requests coming up create new issue(s), please.

**Pull request merged**